internal/peer/snapshot/submitrequest.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package snapshot
+
+import (
+	"context"
+	"fmt"
+
+	pb "github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/hyperledger/fabric/bccsp"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+// submitRequestCmd returns the cobra command for snapshot submitrequest command
+func submitRequestCmd(clientConnection *ClientConnection, cryptoProvider bccsp.BCCSP) *cobra.Command {
+	snapshotSubmitRequestCmd := &cobra.Command{
+		Use:   "submitrequest",
+		Short: "Submit a snapshot request for the given block number.",
+		Long:  "Submit a snapshot request for the given block number. When the blockNumber parameter is set to 0 or not provided, it will submit a snapshot request for the current block number.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return submitRequest(cmd, args, clientConnection, cryptoProvider)
+		},
+	}
+
+	flagList := []string{
+		"channelID",
+		"blockNumber",
+		"peerAddress",
+		"tlsRootCertFile",
+	}
+	attachFlags(snapshotSubmitRequestCmd, flagList)
+
+	return snapshotSubmitRequestCmd
+}
+
+func submitRequest(cmd *cobra.Command, args []string, cc *ClientConnection, cryptoProvider bccsp.BCCSP) error {
+	if err := validateSubmitRequest(); err != nil {
+		return err
+	}
+
+	// Parsing of the command line is done so silence cmd usage
+	cmd.SilenceUsage = true
+
+	// create client connection if not provided
+	if cc == nil {
+		var err error
+		cc, err = NewClientConnection(cmd.Name(), cryptoProvider)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Just to take a note, here we have to compromise on the readability of the production code for testing with the mocks, primarily because of the flags. Looking at this code, one may wonder, who else would be creating the client (especially, the parameters of this function both are present, the `*ClientConnection` and the `cryptoProvider`)... not sure what is a cleaner way but that's how it seems to be in the existing code for the other commands as well.